components/tidb_query_datatype/src/codec/mysql/enums.rs

```diff
@@ -0,0 +1,71 @@
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Enum {
+    data: Vec<u8>,
+    offset: Vec<usize>,
+    value: usize,
+}
+
+impl Enum {
+    fn get(&self, idx: usize) -> &[u8] {
+        assert!(idx < self.offset.len());
+
+        let start = self.offset[idx];
+        let end = if idx < self.offset.len() - 1 {
+            self.offset[idx + 1]
+        } else {
+            self.offset.len()
+        };
+
+        &self.data[start..end]
+    }
+}
+
+impl ToString for Enum {
+    fn to_string(&self) -> String {
+        let buf = Vec::from(self.get(self.value));
+
+        unsafe { String::from_utf8_unchecked(buf) }
```

Comment:
Not sure if this is the correct way of doing this. I think generate string like `\x00\x30\"` is a better idea. What do you think? @breeswish

Reply:
Do you mean using `from_utf8_lossy` ?

Reply:
Using `String::from_raw_parts` to build string directly？

Reply:
Personally, I prefer to use `from_utf8_lossy` for now, and see where `ToString` is called, to avoid using unsafe functions. We can add `TODO:` here and modify this later when we are sure about the requirements and intentions of `ToString` of these types.

Reply:
OK